src/HotChocolate/Core/src/Execution/Processing/SubscriptionExecutor.Subscription.cs

```diff
@@ -202,6 +205,9 @@ private async ValueTask<ISourceStream> SubscribeAsync()
                         await rootSelection.Field.SubscribeResolver!.Invoke(middlewareContext)
                             .ConfigureAwait(false);
 
+                    // mark operation context as Completed in order to free all associated resources and allow to return it to pool
+                    ((IExecutionTaskContext)operationContext).Completed();
```

Comment:
Does it really complete?

The task context is for internal execution tasks to mark their work as completed. The issue here is that it will rain the completed count by one and the compare completed to enqueued tasks which would fail and block completion. The context would then be abandoned.

There is one more use case in this scenario I worry about. I will have a look at this particular piece. In the our issue case here we want to mark this operation as completed since we no it actually has not pending work.

I will have a look at this in the morning.

side-note for myself :)

In the case we have pending work and we need to abandon the context to to errors we need to ensure that we still clean and unsubscribe.